packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -313,19 +313,103 @@ class DateTimePicker extends React.Component {
 
   handleDateChangeRaw(event, value) {
     this.dateValue = value;
+    const validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
+    if (!validDate) {
+      this.setState({
+        dateTime: null,
+      });
+    }
     this.handleChangeRaw(event, value);
   }
 
   handleTimeChange(event, time) {
     this.timeValue = time;
-    const validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
+    let validDate = DateUtil.isValidDate(this.dateValue, this.state.dateFormat) && this.isDateTimeAcceptable(DateTimeUtils.convertDateTimeStringToMomentObject(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds, this.initialTimeZone));
     const validTime = DateTimeUtils.isValidTime(this.timeValue, this.props.showSeconds);
-    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    let previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+    let dateChanged = false;
+
+    if (event.type === 'keydown') {
+      // Keyboard shortcut is pressed, split the time value into hour and minute inputs
+      const timeArray = this.timeValue.split(':');
+      if (event.key === 'n' || event.key === 'N') {
+        // `N` sets Time to now, so Date should be set to today
+        previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+        dateChanged = true;
+      } else if (event.key === '-' || event.key === '_') {
+        // `-` subtracts 1 minute`from time, so check date to see if it needs changed
+        if (!validDate) {
+          // Date is not valid, set date to today making it valid
+          previousDateTime = DateUtil.createSafeDate(DateUtil.getCurrentDate(), this.initialTimeZone);
+          dateChanged = true;
+        }
+
+        if (timeArray[0] === '23' && timeArray[1] === '59') {
+          // Time is 23:59 which means previous datetime should be on the wrong day, lower it by 1
+          previousDateTime = previousDateTime.subtract(1, 'd');
+          dateChanged = true;
+        }
+
+        const changedDateTime = DateTimeUtils.updateTime(previousDateTime, this.timeValue, this.props.showSeconds);
+        const checkHour = previousDateTime.clone().subtract(1, 'minute').hours();
```

Comment:
For Daylight Savings handling when DST begins, the keyboard shortcuts for `+` & `-` are performing as expected when focus is in the time input, correctly skipping the missing hour when incrementing and decrementing time (`01:59`+`+`=`03:00`, `03:00`+`-`=`01:59`), but when focus is in the date input, the time is not being handled correctly.

The current behavior when the date is any day before or after when DST begins, (e.g. `March 06, 2020`) and the time is between 02:00 and 03:00, valid on March 6th (e.g. `03/06/2020 02:30`), with focus in the date input, pressing `+` these number counts, changes the value to:
1. `03/07/2020 02:30` (correct)
2. `03/08/2020 03:30` (correct)
3. `03/09/2020 03:30` (incorrect)
4. `03/10/2020 03:30` (incorrect)

And same is true in reverse starting with `03/12/2020 02:30` with pressing `-`, the time is correct on `03/08/2020` at `03:30`, but should go back to `02:30` starting on `03/07/2020`:

![incorrect-DST-start-missing-hour-and-hotkeys-PR#1520](https://user-images.githubusercontent.com/1355485/160500856-317c5479-33a8-4700-aeac-16614c72b086.gif)

---

The correct expected behavior is when starting with the time between 02:00 and 03:00 on the days before/after the start of DST (e.g. `03/06/2020 02:30`), for time to shift up an hour on the day of the start of DST when the missing hour occurs, but then go back to the original time for days before or after when the missing hour does not occur. Using the same example above when pressing `+` these number counts, the correct values would be:
1. `03/07/2020 02:30` (correct)
2. `03/08/2020 03:30` (correct)
3. **`03/09/2020 02:30`** (correct)
4. `03/10/2020 02:30` (correct)
5. `03/11/2020 02:30` (correct)

---

Reply:
Approving current state and leaving this bug fix for a future enhancement to resolve. https://github.com/cerner/terra-framework/pull/1520#pullrequestreview-923876182